app.py

```diff
@@ -44,8 +50,87 @@ def about():
 
 @app.route("/home")
 @login_required
+# current page is login required, which means not logging will be redirected
 def home():
-    return render_template('home.html')
+    context = {"username": current_user.username}
+    return render_template('home.html', **context)
+
+
+@app.route("/home", methods=["POST"])
+@login_required
+def upload_strategy():
+
+    if "user_file" not in request.files:
+        return "No user_file is specified"
+    if "strategy_name" not in request.form:
+        return "Strategy name may not be empty"   
+    file = request.files["user_file"]
+    name = request.form["strategy_name"] 
+    '''
+        These attributes are also available
+
+        file.filename               # The actual name of the file
+        file.content_type
+        file.content_length
+        file.mimetype
+
+    '''
+    if file.filename == "":
+        return "Please select a file"
+
+    if not allowed_file(file.filename):
+        return "Your file extension type is not allowed"
+
+    if not file:
+        return "File not found. Please upload it again"
+
+    username = current_user.username
+    folder = 'strategies/' + username
+    if not os.path.exists(folder):
+        os.makedirs(folder)
+
+    # get the number of folders
+
+    cnt = len([_ for _ in os.listdir(folder)])
+
+    print("folder number is ", cnt)
+    new_folder = "strategy" + str(cnt + 1)
+    strategy_folder = os.path.join(folder, new_folder)
+    os.makedirs(strategy_folder)  # it must be new file
+
+    # name file to be main.py
+    filepath = os.path.join(strategy_folder, "main.py")
+    file.save(filepath)
+    result = Run([filepath], do_exit=False)
+
+    # may be need threshold
+    if "global_note" not in result.linter.stats or \
+            result.linter.stats['global_note'] <= 0:
+        shutil.rmtree(strategy_folder)
+        return "Your strategy has error or is not able to run! \
```

Comment:
can we please show what error it is on the U.I.?

Reply:
Do you mean to show it in the same page instead of returning a sentence ?